### PR TITLE
JDK-8328225: Convert ImageDecoratedDnD.html applet test to main

### PR DESCRIPTION
**Pull request by honkar-jdk**
Following test is converted to main using PassFailJFrame.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328225](https://bugs.openjdk.org/browse/JDK-8328225): Convert ImageDecoratedDnD.html applet test to main (**Bug** - P4)

### Reviewers
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Committer) ⚠️ Review applies to [7e35e23d](https://git.openjdk.org/jdk/pull/18315/files/7e35e23dc503f6ecc3fe94e533c0ab3425bd1f7f)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**) ⚠️ Review applies to [1a565eae](https://git.openjdk.org/jdk/pull/18315/files/1a565eae04c7e90898e0913c05724cdf9fe3b83b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18315/head:pull/18315` \
`$ git checkout pull/18315`

Update a local copy of the PR: \
`$ git checkout pull/18315` \
`$ git pull https://git.openjdk.org/jdk.git pull/18315/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18315`

View PR using the GUI difftool: \
`$ git pr show -t 18315`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18315.diff">https://git.openjdk.org/jdk/pull/18315.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18315#issuecomment-1998604902)

### CODE REVIEW

File: test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java [hunk 1]
```diff
@@ -21,177 +21,79 @@
...
+
+            2. When the mouse enters the red panel during the drag, the panel
+               should turn yellow.
+                On the systems that supports pictured drag, the image under the drag-cursor
+                should appear (anchor is shifted, from top-left corner of the picture to
+                inside the picture to 10pt in both dimensions).
+
+                In WIN32 systems the image under cursor would be visible ONLY over
+                the drop targets with activated extended OLE D'n'D support (that are,
+                the desktop and IE).
```
Comment (honkar-jdk) [inline test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java:56]: @prrace Do we retain the testing instructions related to Win32 as they are quite old?

Comment (aivanov-jdk) [reply test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java:56]: Win32 is not that old… Java still uses these APIs in 64-bit world.

I haven't run the test myself (yet). I suggest replacing _Win32_ with _Windows_. Is image under cursor visible when you drag'n'drop on Windows? It should be.

Comment (honkar-jdk) [reply test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java:56]: Case 1: A green window below the anchor + circle is seen when CTRL + Mouse is used to perform DnD.
Case 2: A circle + anchor is seen (without a green window) when just Mouse is used to perform DnD.

https://github.com/openjdk/jdk/assets/95945681/4ed2aa47-611c-4957-b930-b86732a95f76

https://github.com/openjdk/jdk/assets/95945681/d8990f71-e90f-4ee2-a7b2-ba5458fbd879

Comment (aivanov-jdk) [reply test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java:56]: You can see the image under the cursor. The image appears when you hover over AWT windows, Firefox window but the image disappears when you hover over Cygwin terminal window.

The image is the translucent blue rectangle with red circle.

Comment (honkar-jdk) [reply test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java]: @aivanov-jdk Updated the instructions to make it more clear. Can you please re-review?

File: test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java [hunk 2]
```diff
@@ -21,177 +21,84 @@
...
+
+            1. Click on the button and drag to the red panel by pressing
+               the "CTRL" key on the keyboard.
+
+            2. When the mouse enters the red panel during the drag, the panel
+               should turn yellow.
+
+                On the systems that supports pictured drag, the image under the drag-cursor
+                should appear.
+                "Image under drag-cursor" is a translucent blue rectangle + red circle
```
Comment (DamonGuy) [inline test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java:53]:
```suggestion
                On the systems that supports pictured drag, the image under the 
                drag-cursor should appear.
                "Image under drag-cursor" is a translucent blue rectangle + red circle
```

In the instructions window, there's an extra newline due to exceeding the character limit per line.

**Review by DamonGuy — CHANGES_REQUESTED** [review]
Do we need to add any instructions regarding the blue "empty" boxes that appear?

**Comment by honkar-jdk** [comment]
> Do we need to add any instructions regarding the blue "empty" boxes that appear?

Since we are checking the second case i.e CTRL + Mouse Drag, the empty boxes don't appear hence I skipped it.

File: test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java [hunk 3]
```diff
@@ -21,177 +21,84 @@
...
+ * @run main/manual ImageDecoratedDnD
+ */
 
+public class ImageDecoratedDnD {
+    private static final String INSTRUCTIONS = """
+            When test runs a Frame which contains a yellow button labeled "Drag ME!"
+            and a RED Panel will appear.
+
+            1. Click on the button and drag to the red panel by pressing
+               the "CTRL" key on the keyboard.
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java:46]:
```suggestion
            1. Click on the button and drag it to the red panel while holding
               the "CTRL" key on the keyboard.
```

File: test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java [hunk 4]
```diff
@@ -21,177 +21,84 @@
...
+
+                On the systems that supports pictured drag, the image under the drag-cursor
+                should appear.
+                "Image under drag-cursor" is a translucent blue rectangle + red circle
+                and includes an anchor that is shifted from top-left corner of the
+                picture to inside the picture to 10pt in both dimensions.
+
+                On Windows system the image under cursor would be visible ONLY over
+                the drop targets with activated extended OLE D'n'D support (that are,
+                the desktop and IE).
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java]:
```suggestion
                On Windows, the image under the cursor would be visible ONLY over
                drop targets with activated extended D'n'D support. It means the image
                may not be displayed when dragging over some windows, it is not error. The
                image should be display when dragging over the red/yellow panel.
```
Let's clarify the instructions a bit.

File: test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java [hunk 5]
```diff
@@ -21,177 +21,84 @@
...
+
+            3. Release the mouse button.
+
+                The panel should turn red again and a yellow button labeled ,
+                "Drag ME!" should appear inside the panel. You should be able,
+                to repeat this operation multiple times.
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java]:
```suggestion
                The panel should turn red again and a yellow button labeled
                "Drag ME!" should appear inside the panel. You should be able
                to repeat this operation multiple times.
```
No comma here.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

**Review by DamonGuy — APPROVED** [review]
Makes sense. Confirmed OK with CTRL.

File: test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java (hunk 3 above)
Comment (honkar-jdk) [reply test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java:46]: Updated.

File: test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java [hunk 6]
```diff
@@ -21,177 +21,85 @@
...
-    public void start() {
-        Frame f = new Frame("Use keyboard for DnD change");
+    public static Frame createUI() {
+        Frame frame = new Frame("Ctrl + Drag - Image DnD test");
         Panel mainPanel;
         Component dragSource, dropTarget;
 
-        f.setBounds(0, 400, 200, 200);
-        f.setLayout(new BorderLayout());
+        frame.setBounds(0, 400, 400, 400);
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java:89]:
```suggestion
        frame.setSize(400, 400);
```
The frame is positioned by `PassFailJFrame`, so setting the location is redundant.

File: test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java [hunk 7]
```diff
@@ -42,27 +42,29 @@ public class ImageDecoratedDnD {
...
                 in both dimensions.
 
-                On Windows system the image under cursor would be visible ONLY
-                over the drop targets with activated extended OLE DnD support
-                (that are, the desktop and IE).
+                On Windows, the image under the cursor would be visible ONLY over
+                drop targets with activated extended D'n'D support.
+                It means the image may not be displayed when dragging over some
+                windows, it is not error.
+                The image should be display when dragging over the red/yellow panel.
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java:62]:
```suggestion
                windows, it is not an error.
                The image should be displayed when dragging over the red/yellow panel.
```

File: test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java [hunk 8]
```diff
@@ -31,7 +31,7 @@
 /*
  * @test
  * @bug 4874070
- * @summary Tests basic DnD functionality
+ * @summary Tests CTRL + DnD functionality.
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/dnd/ImageDecoratedDnD/ImageDecoratedDnD.java:34]: I think summaries look better without the full stop in the end…

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**